File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/UniversidadBO.java
package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.localidad.model.Universidad;
import java.rmi.Remote;
import java.rmi.RemoteException;

public interface UniversidadBO extends Remote{
    
    public Integer insertar(String nombre,String direccion,String correoPersonalAdministrativo,
                            String extensionDominioCorreo,Integer idPoliticaRegular,Integer idConsorcioPerteneciente) throws RemoteException;
    
    public Integer modificar(Integer idUniversidad,String nombre,String direccion,
                             String correoPersonalAdministrativo,String extensionDominioCorreo,Integer idPoliticaRegular,
                             Integer idConsorcioPerteneciente) throws RemoteException;
    
    public Integer eliminar(int idUniversidad) throws RemoteException;
    
    public ArrayList<Universidad> listarTodos(Integer idPoliticaRegular,Integer idConsorcioPerteneciente, Integer limite) throws RemoteException;
    
    public Universidad obtenerPorId(int idUniversidad) throws RemoteException;
    
    public ArrayList<Universidad> listarTodos_UniversidadesDeUsuarioPorID(Integer idUsuario) throws RemoteException;
    
    public ArrayList<Universidad> listarTodos_UniversidadesDeUsuario(Integer idUsuario,String nombres, String apellidos,
                                                                      String correoInstitucional,String nombre_usuario,String fechaRegistro,
                                                                      String tipoUsuario,Integer idPoliticaRegular,Integer idConsorcioPerteneciente) throws RemoteException;
    
    public Integer eliminarUniversidadesDeUsuario(Integer idUsuario) throws RemoteException;
    
}

File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaUniversidadBO/UniversidadBO.cs
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftBibliotecaUniversidadBO
{
    public class UniversidadBO : BaseBO
    {
        public int insertar(string nombre, string direccion, string correoPersonalAdministrativo, string extensionDominioCorreo, int idPoliticaRegular, int idConsorcioPerteneciente)
        {
            return this.WsCliente.universidad_insertar(nombre, direccion, correoPersonalAdministrativo, extensionDominioCorreo, idPoliticaRegular, idConsorcioPerteneciente);
        }

        public int modificar(int idUniversidad, string nombre, string direccion, string correoPersonalAdministrativo, string extensionDominioCorreo, int idPoliticaRegular, int idConsorcioPerteneciente)
        {
            return this.WsCliente.universidad_modificar(idUniversidad, nombre, direccion, correoPersonalAdministrativo, extensionDominioCorreo, idPoliticaRegular, idConsorcioPerteneciente);
        }

        public int eliminar(int idUniversidad)
        {
            return this.WsCliente.universidad_eliminar(idUniversidad);
        }


        public BindingList<universidad> listarTodos(int idPoliticaRegular, int idConsorcioPerteneciente, int limite)
        {
            try
            {
                universidad[] universidades = this.WsCliente.universidad_listarTodos(idPoliticaRegular, idConsorcioPerteneciente, limite);
                if (universidades == null) return new BindingList<universidad>();
                return new BindingList<universidad>(universidades);
            }
            catch (Exception e)
            {
                return new BindingList<universidad>();
            }
        }

        public universidad obtenerPorId(int idUniversidad)
        {
            return this.WsCliente.universidad_obtenerPorId(idUniversidad);
        }

        public BindingList<universidad> listarTodos_UniversidadesDeUsuarioPorID(int idUsuario)
        {
            try
            {
                universidad[] universidades = this.WsCliente.listarTodos_UniversidadesDeUsuarioPorID(idUsuario);
                if (universidades == null) return new BindingList<universidad>();
                return new BindingList<universidad>(universidades);
            }
            catch (Exception e)
            {
                return new BindingList<universidad>();
            }
        }

        public BindingList<universidad> listarTodos_UniversidadesDeUsuario(int idUsuario, string nombres, string apellidos,
                                                                          string correoInstitucional, string nombre_usuario, string fechaRegistro,
                                                                          string tipoUsuario, int idPoliticaRegular, int idConsorcioPerteneciente)
        {
            try
            {
                universidad[] universidades = this.WsCliente.listarTodos_UniversidadesDeUsuario(idUsuario, nombres, apellidos, correoInstitucional, nombre_usuario, fechaRegistro, tipoUsuario, idPoliticaRegular, idConsorcioPerteneciente);
                if (universidades == null) return new BindingList<universidad>();
                return new BindingList<universidad>(universidades);
            }
            catch (Exception e)
            {
                return new BindingList<universidad>();
            }
        }
        public int eliminarUniversidadesDeUsuario(int idUsuario)
        {
            return this.WsCliente.eliminarUniversidadesDeUsuario(idUsuario);
        }


    }
}
